db/system_distributed_keyspace.cc

```diff
@@ -567,14 +567,30 @@ system_distributed_keyspace::get_cdc_desc_v1_timestamps(context ctx) {
     co_return res;
 }
 
+static std::optional<lowres_clock::duration> get_duration(const cql3::untyped_result_set_row&row, std::string_view col_name) {
+    auto dur_opt = row.get_opt<cql_duration>(col_name);
+    if (!dur_opt) {
+        return std::nullopt;
+    }
+    return std::chrono::duration_cast<lowres_clock::duration>(std::chrono::nanoseconds(dur_opt->nanoseconds));
+};
+
 future<qos::service_levels_info> system_distributed_keyspace::get_service_levels() const {
     static sstring prepared_query = format("SELECT * FROM {}.{};", NAME, SERVICE_LEVELS);
 
     return _qp.execute_internal(prepared_query, db::consistency_level::ONE, internal_distributed_query_state(), {}).then([] (shared_ptr<cql3::untyped_result_set> result_set) {
         qos::service_levels_info service_levels;
         for (auto &&row : *result_set) {
             auto service_level_name = row.get_as<sstring>("service_level");
-            qos::service_level_options slo{};
+            qos::service_level_options slo{
+                .read_timeout = get_duration(row, "read_timeout"),
+                .write_timeout = get_duration(row, "write_timeout"),
+                .range_read_timeout = get_duration(row, "range_read_timeout"),
+                .counter_write_timeout = get_duration(row, "reacounter_write_timeoutd_timeout"),
```

Comment:
The literal string in the end of the line got mangled.

Reply:
Ugh. Now I like the idea of unifying all these timeouts into a single per-service-level timeout even more.